hudi-flink-datasource/hudi-flink1.18.x/src/main/java/org/apache/hudi/adapter/HoodieHiveCatalogAdapter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.adapter;
+
+import org.apache.hudi.internal.schema.InternalSchema;
+
+import org.apache.flink.table.catalog.Catalog;
+import org.apache.flink.table.catalog.CatalogBaseTable;
+import org.apache.flink.table.catalog.ObjectPath;
+import org.apache.flink.table.catalog.TableChange;
+import org.apache.flink.table.catalog.exceptions.CatalogException;
+import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+
+import java.util.List;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Adapter clazz for {@code HoodieHiveCatalog}.
+ */
+public interface HoodieHiveCatalogAdapter extends Catalog, TableSchemaSupplier, FlinkTypeConverter {
+
```

Comment:
Not sure why we need these 2 classes: TableSchemaSupplier, FlinkTypeConverter, should they both be pure utilities? Do they have different impls for different Flink releases? Even though we can still use it as static utilities class.

Reply:
@danny0405 I agree with you. 
However, based the current dependencies between flink datasource modules, the class in hudi-flink1.X module could not introduce classes in hudi-flink mdoule. Therefore, I have temporarily added `TableSchemaSupplier `, `FlinkTypeConverter ` for now.

Reply:
@danny0405 I've updated the pr to move new `alterTable` methd into `HoodieHiveCatalog`